torch/csrc/jit/script/compiler.cpp

```diff
@@ -229,10 +229,14 @@ struct Environment {
     return std::make_shared<SimpleValue>(load->output());
   }
 
-  void insertStore(const std::string& name, const SourceRange& loc, Value* v) {
+  void insertStore(
```

Comment:
Maybe add a comment demonstrating a case when we need to use type different than `store->input()->type()`.